### PR TITLE
6 | Palindrome | v1.0

### PR DESCRIPTION
**Pull request by hugocxl**
Here's my submission for the Palindrome checker assignment: https://www.essentialist.dev/products/the-software-essentialist/categories/2152752280/posts/2166918438

Glad to hear your thoughts!

### CODE REVIEW

File: 1_The_Physical/Getting_Started_Emergent_Design/1_Red_Green_Refactor/1_2_Palindrome/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export class PalindromeChecker {
+  private static reverseString(str: string) {
+    return str.split('').reverse().join('')
+  }
+
+  public static check(str: string) {
+    const lowerCaseStr = str.toLowerCase().replace(/\s/g, '')
+    const reversedString = this.reverseString(lowerCaseStr)
+
+    if (reversedString === lowerCaseStr) return true
```
Comment (rochaalexandre) [inline 1_The_Physical/Getting_Started_Emergent_Design/1_Red_Green_Refactor/1_2_Palindrome/src/index.ts:10]: You could simplify this code by returning the result of the if conditional directly.
return ` return reversedString === lowerCaseStr)`

**Review by rochaalexandre — COMMENTED** [review]
Pretty good commits flow, you started simpled and progressed test by test, removed the duplication after sometime and also refactor to simplify the code once you felt it could be improved.

File: 1_The_Physical/Getting_Started_Emergent_Design/1_Red_Green_Refactor/1_2_Palindrome/src/index.ts (hunk 1 above)
Comment (hugocxl) [reply 1_The_Physical/Getting_Started_Emergent_Design/1_Red_Green_Refactor/1_2_Palindrome/src/index.ts]: @rochaalexandre totally 🤦‍♂️

File: 1_The_Physical/Getting_Started_Emergent_Design/1_Red_Green_Refactor/1_2_Palindrome/src/index.spec.ts [hunk 2]
```diff
@@ -1,4 +1,33 @@
...
+
+  test.each([
+    { value: 'bill', expected: false },
+    { value: 'cat', expected: false },
+    { value: 'mom', expected: true },
+    { value: 'Wow', expected: true },
+    { value: 'Was It A Rat I Saw', expected: true },
+    { value: 'Not a palindrome', expected: false },
+  ])(
+    'when checking "$value" it should return "$expected"',
```
[the remark this answers is not in the document]
Comment (hugocxl) [reply 1_The_Physical/Getting_Started_Emergent_Design/1_Red_Green_Refactor/1_2_Palindrome/src/index.spec.ts]: @rochaalexandre That is great advice right there. I reviewed my tests declarations to formulate the cases around functionality, avoiding technical descriptions.

**Comment by hugocxl** [comment]
@rochaalexandre Thanks for your review mate! As always, valuable advice. I checked your Palindrome assignment for guidance and I learnt a lot from that. Great job there. Added some improvements to my exercise based on that. Thanks for taking the time.

**Pull request merged**